### PR TITLE
Update cf-icons to use inline svgs rather than an icon font

### PR DESCRIPTION
**Pull request by jimmynotjim**
Icon fonts have some shortcomings with accessibility and
maintainability. To simplify usage, we've converted them to inline SVGs
that are then included by each project's templating system. The process
from hand-off from Graphic Designers requires removing left/right
whitespace (we batch processed the icons through
an Illustrator script to save hours of work), followed by optimization
via SVGO within Capital Framework. The resulting icons are then copied
to their respective projects via that project's build process. Finally,
we updated each of the other components that require icons to use the
new pattern. The exception is the example for the rotating icon within
a button (see the usage doc for further info).

## Additions

- SVG Icons
- CSS to size, position and color icons

## Changes

- Converts other CF Components to use the new SVG Icons

## Testing

- Follow the [CONTRIBUTING.md instructions for testing locally](https://github.com/cfpb/capital-framework/blob/canary/CONTRIBUTING.md#testing-components-locally). The site specific branch is `svg-icons-test`
- In `svg-icons-test` run gulp build && npm start
- Navigate to `http://localhost:3000/components/cf-icons/` and read the documentation and review the tables of icons
- Navigate through the rest of the components and ensure icons are working as expected where they are currently used (compare with the live docs for reference).

## Screenshots

screen shot 2018-02-06 at 2 01 51 pm

screen shot 2018-02-06 at 2 02 22 pm

screen shot 2018-02-06 at 2 02 45 pm

screen shot 2018-02-06 at 2 03 01 pm

## Notes

- There are 400 svg files (200 base file + 200 aliases), you can safely ignore them when reviewing the changed files.
- Due to the fact that non-icon components would stop working after being updated to use the new SVG icon patterns within each of them if a user didn't update cf-icons as well. Scott and I are proposing that we merge this PR to a pre-release branch, and publish that as a pre-release on NPM. We'd then update cfgov-refresh to use the pre-release to give us a month or so to ensure we've worked out any issues.

## Todos

- Update the necessary icon source files (update.svg) to include `-spin` modifiers with the animation code embedded in the SVG.

## Checklist

* [x] Changes are limited to a single goal (no scope creep)
* [x] Code can be automatically merged (no conflicts)
* [x] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [x] Passes all existing automated tests
* [x] New functions include new tests
* [x] New functions are documented (with a description, list of inputs, and expected output)
* [x] Placeholder code is flagged
* [x] Visually tested in supported browsers and devices
* [x] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG) SEE NOTES SECTION

### Browsers

- [x] Chrome
- [x] Safari
- [x] FF
- [x] IE10
- [x] IE9
- [x] IE8
- [x] iOS
- [ ] Android

### Accessibility

- [x] Keyboard friendly
- [x] Screen reader friendly

### Other

- [x] Is useable without CSS
- [x] Is useable without JS
- [x] Flexible from small to large screens
- [x] No linting errors or warnings
- [x] JavaScript tests are passing

### CODE REVIEW

File: src/cf-buttons/src/atoms/buttons-with-icons.less [hunk 1]
```diff
@@ -14,9 +21,9 @@
     padding-left: unit( 11px / @btn-font-size, em );
     border-left: 1px solid mix( @btn-bg, @btn-text, 50% );
     margin-left: unit( 7px / @btn-font-size, em );
+    width: 18px;
```
Comment (Scotchester) [inline src/cf-buttons/src/atoms/buttons-with-icons.less]: em value?

File: src/cf-icons/usage.md [hunk 2]
```diff
@@ -117,6 +117,14 @@ we set `vertical-align: text-top;`.
 Finally, setting `fill: currentColor;` tells the SVG to set its path's fill
 color to match the `color` value of its parent element.
 
+#### Caveats
+
+There are two modifications based on restrictions in Internet Explorer 8 and 9.
+
+First, IE8 does not support `fill: currentColor`. Typically the fallback would be to use a png image, but due to the inability to know what the background or text color of it's surroundings are, we found it better to fall back to the paired text with no icon.
+
+Second, IE9 displays svgs as full width by default (not the paths, just the SVG container). To eliminate this issue we've set the width of the SVG to match the height. The whitespace to the left or right may not be quite accurate, but we determined this is an acceptable difference for a legacy browser like IE9.
```
Comment (Scotchester) [inline src/cf-icons/usage.md]: Trivial: `SVGs`. And wrap these two paragraphs at 80 characters.

File: src/cf-buttons/src/atoms/buttons-with-icons.less (hunk 1 above)
Comment (jimmynotjim) [reply src/cf-buttons/src/atoms/buttons-with-icons.less]: Oh shoot, that shouldn't be there, left it in from testing.

File: package.json [hunk 3]
```diff
@@ -26,7 +26,8 @@
     "build": "gulp build",
     "test": "gulp test",
     "cf-link": "node scripts/npm/link",
-    "postinstall": "npm run cf-link"
+    "postinstall": "npm run cf-link",
+    "process-icon-svgs": "svgo -o src/cf-icons/src/icons --enable=addClassesToSVGElement --enable=removeStyleElement --enable=removeAttrs --config='{ \"plugins\": [ { \"addClassesToSVGElement\": { \"className\": \"cf-icon-svg\" } }, { \"removeAttrs\": { \"attrs\": [ \"fill\", \"path:class\", \"circle:class\" ] } }, { \"cleanupIDs\": { \"force\": \"true\" } } ] }'"
```
Comment (anselmbradford) [inline package.json]: What calls this? I don't see any other reference to this script?

File: src/cf-forms/src/molecules/btn-inside-input.less [hunk 4]
```diff
@@ -14,5 +18,16 @@
         // Set the right and top distances to match typical button padding.
         right: unit( @btn-h-padding / @btn-font-size, em );
         top: unit( @btn-v-padding / @btn-font-size, em );
+
+        // TODO: Double check the colors here with designers
```
Comment (anselmbradford) [inline src/cf-forms/src/molecules/btn-inside-input.less]: Probably something that could be done before the PR goes in.

File: src/cf-icons/src/cf-icons.less [hunk 5]
```diff
@@ -7,21 +7,65 @@
...
+    fill: currentColor;
+
+    .lt-ie10 & {
+        // IE 9 require a width otherwise the SVG takes up 100%
+        width: @cf-icon-height;
+    }
+
+    .lt-ie9 & {
+        // IE 8 doesn't support currentColor, hide icons and let the paired
+        // text stand on it's own
```
Comment (anselmbradford) [inline src/cf-icons/src/cf-icons.less]: TRIVIAL: `its own.`

File: package.json (hunk 3 above)
Comment (Scotchester) [reply package.json]: It's called manually when someone adds/updates icons. We need to add documentation about that somewhere.

**Pull request merged**